platform/engineering/request-for-comment/tracking-submissions-from-origin-to-destination.md

```diff
@@ -0,0 +1,399 @@
+
+const id = await digestMessage(hashContent);
+// a11af0b512e265c7fffaf6c8dd8ebb9933dc912f87d8c45c62184096fa0a7640
+```
+
+### Other Submissions
+
+What about submissions outside the digital forms? With the IDs being generated using the context of the submission itself, IDs can be determined by the intaking system during processing or digitization.
+
+![Form 20-0995](images/tracking-submissions-from-origin-to-destination/paper-form.png)
+
+In the above example, you can gather the information for the id from the content. The content matches the values in the example above, so you would end up with the same ID!
+
+## Risks
+
+### Types
+
+Having types be global requires any system generating an ID would require that value as context. This can be mitigated by having a globally acknowledged place of record as well as only required the system or origin to be aware of Types it is a source of origin for, however it is still an external context requirement.
+
+### Duplicates
+
+There can be cases where a Veteran submits the same exact submission multiple times. This should be mitigated by the SME selecting data that guarantees uniqueness, however that may not always be the case.
+
+This is an opportunity to deduplicate submissions at the submission process however that may not be the desired outcome as Veterans are allowed to submit duplicates.
+
+In those cases, we could add an optional [salt](https://en.wikipedia.org/wiki/Salt_(cryptography)) value to the context. This could be used if set, but allowed to be empty when no salt is needed.
+
+### Context Changes
+
+If, through the process of submission, values used in the context of the id change, the resulting id would also change.
+
+**Examples**:
+
+#### Format Changes
+
+- Veteran submits evidence as a PNG file to VA.org.
+- That evidence is sent to Lighthouse where it is converted to a PDF file and send to CMP.
+
+If the Submission ID is generated using a hash of the PNG file as context, that context would be lost in the conversion to PDF.
+
+#### Context Loss
+
+- Veteran submits data for a form that is too long to fit in the resulting form area
+- When the PDF form is generated, some of the data is cut off or moved to an alternate section
+
+If the Submission ID is generated using that full data, the context will have been changed thus changing the ID.
+
+## Alternatives
+
+### Random IDs
+
+Assign a random unique ID for each submission. Assignment is much easier at origin as a properly unique value, such as a UUID, can generated without any context needed.
+
+A downside would be that the ID would need to either be included in the submission content or be purposely carried by each system the submission travels through in the metadata of the requests.
+
+### Central ID Map
+
+Each system in the submission chain generates its own unique ID. Once a submission has moved from SystemA to SystemB, SystemA will report to a central location the corelation of SystemA.id to SystemB.id.
```

Comment:
What about a central polling system which can access the receiving system and query those to determine the existence/status of a submission?